src/java.base/share/classes/java/time/format/DateTimePrintContext.java

```diff
@@ -128,6 +141,33 @@ private static TemporalAccessor adjust(final TemporalAccessor temporal, DateTime
             return temporal;
         }
 
+        // The chronology and zone fields of Formatter are usually null,
+        // so the non-null processing code is placed in a separate method
+        return adjustWithOverride(temporal, overrideChrono, overrideZone);
```

Comment:
The "so..." doesn't explain much, perhaps:
Placing the non-null cases in a separate method allows more flexible code optimizations"